lib/finch/http1/pool.ex

```diff
@@ -74,6 +74,56 @@ defmodule Finch.HTTP1.Pool do
     end
   end
 
+  @impl Finch.Pool
+  def async_request(pool, req, opts) do
+    owner = self()
+
+    pid =
+      spawn(fn ->
```

Comment:
Maybe we should link?

```suggestion
      spawn_link(fn ->
```

This will make sure any unexpected crashes will propagate up.

Reply:
If we link, there may be no need to monitor, as this process we crash. Alternatively we link, trap exits, and monitor.

Reply:
Yep, that would definitely simplify things. I was thinking that we wouldn’t want internal crashes to take down the caller, only the caller to take down the request, but the request crashing already means the caller isn’t going to receive messages it’s probably expecting, so it’s probably better to just crash.

Reply:
@josevalim See 9ae43ef7f71aef81583e2ff1ef74512613ce788d -- do you have any insight into why the `spawn_link` process doesn't immediately terminate when the caller does? Specifically, [this test](https://github.com/sneako/finch/pull/228/files#diff-0ce8331b6dbdaa98b49a795e62340ca7bb472824c919d4fafb94936967688745R102) was failing when I stripped out the monitoring code. I also tried adding `Process.flag(:trap_exit, true)` and then changing `process_down?(monitor)` to a `should_exit()` that looks for an `{:EXIT, _, _}` message, but didn't seem to receive one.

Reply:
In this code:

```elixir
        spawn(fn ->
          ref =
            Finch.build(:get, url <> "/stream/5/500")
            |> Finch.async_request(finch_name)

          send(outer, ref)
        end)
```

the spawned process immediately exits with reason `:normal` after `send/2`, which does not lead to any broken link. So by the time you send it a exit signal, it is already gone. Adding a `Process.sleep(:infinity)` should fix it.

It it up to you to decide if we still need monitoring. If you start something to stream asynchronously and you never consume it, what does it mean?

Reply:
> the spawned process immediately exits with reason `:normal` after `send/2`, which does not lead to any broken link. So by the time you send it a exit signal, it is already gone. Adding a `Process.sleep(:infinity)` should fix it.

Oh, duh! Thank you. 

I’ll fix that and see how it behaves without monitoring and decide from there.

Reply:
Split the test to explicitly test both normal and abnormal caller exits. Decided to keep the monitoring around because I think canceling the request if there is no caller to receive it is the right thing to do. I'll document this behavior in `async_request/3`, so that folks know that the caller must stick around for the request to continue. I can imagine some potentially useful patterns around this behavior, e.g.

```elixir
Task.async(fn ->
  some_streamed_request()
  |> Enum.find(fn
    {:data, data} -> some_chunk_im_looking_for?(data)
    _ -> false
  end)
end)
```